test/helpers/templates.go

```diff
@@ -0,0 +1,176 @@
+//
+// Copyright 2020 IBM Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package helpers
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/IBM/operand-deployment-lifecycle-manager/pkg/apis/operator/v1alpha1"
+	"github.com/IBM/operand-deployment-lifecycle-manager/test/config"
+)
+
+// newOperandConfigCR is return an OperandConfig CR object
+func newOperandConfigCR(name, namespace string) *v1alpha1.OperandConfig {
+	return &v1alpha1.OperandConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v1alpha1.OperandConfigSpec{
+			Services: []v1alpha1.ConfigService{
+				{
+					Name: "etcd",
+					Spec: map[string]runtime.RawExtension{
+						"etcdCluster": {Raw: []byte(`{"size": 1}`)},
+					},
+				},
+				{
+					Name: "jenkins",
+					Spec: map[string]runtime.RawExtension{
+						"jenkins": {Raw: []byte(`{"service":{"port": 8081}}`)},
+					},
+				},
+				{
+					Name: "jaeger",
+					Spec: map[string]runtime.RawExtension{},
+				},
+			},
+		},
+	}
+}
+
+// newOperandConfigCR is return an OperandRegistry CR object
+func newOperandRegistryCR(name, namespace string) *v1alpha1.OperandRegistry {
+	return &v1alpha1.OperandRegistry{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v1alpha1.OperandRegistrySpec{
+			Operators: []v1alpha1.Operator{
+				{
+					Name:            "etcd",
+					Namespace:       config.TestNamespace1,
+					InstallMode:     "cluster",
+					SourceName:      "community-operators",
+					SourceNamespace: "openshift-marketplace",
+					PackageName:     "etcd",
+					Channel:         "clusterwide-alpha",
+				},
+				{
+					Name:            "jenkins",
+					Namespace:       config.TestNamespace1,
+					SourceName:      "community-operators",
+					SourceNamespace: "openshift-marketplace",
+					PackageName:     "jenkins-operator",
+					Channel:         "alpha",
+					Scope:           v1alpha1.ScopePublic,
+				},
+				{
+					Name:            "jaeger",
+					Namespace:       config.TestNamespace2,
+					SourceName:      "community-operators",
+					SourceNamespace: "openshift-marketplace",
+					PackageName:     "jaeger",
+					Channel:         "stable",
+					Scope:           v1alpha1.ScopePublic,
+				},
+			},
+		},
+	}
+}
```

Comment:
Do we need to add an another OperandRegistry and OperandConfig in the same namespace in the templates for the testing?
```
// newOperandRegistryCR2 is return an OperandRegistry CR object
func newOperandRegistryCR2(name, namespace string) *v1alpha1.OperandRegistry {
	return &v1alpha1.OperandRegistry{
		ObjectMeta: metav1.ObjectMeta{
			Name:      name,
			Namespace: namespace,
		},
		Spec: v1alpha1.OperandRegistrySpec{
			Operators: []v1alpha1.Operator{
				{
					Name:            "kubeturbo",
					Namespace:       config.TestNamespace1,
					SourceName:      "community-operators",
					SourceNamespace: "openshift-marketplace",
					PackageName:     "kubeturbo-operator",
					Channel:         "alpha",
				},
			},
		},
	}
}

// newOperandConfigCR2 is return an OperandConfig CR object
func newOperandConfigCR2(name, namespace string) *v1alpha1.OperandConfig {
	return &v1alpha1.OperandConfig{
		ObjectMeta: metav1.ObjectMeta{
			Name:      name,
			Namespace: namespace,
		},
		Spec: v1alpha1.OperandConfigSpec{
			Services: []v1alpha1.ConfigService{
				{
					Name: "kubeturbo",
					Spec: map[string]runtime.RawExtension{
						"kubeturbo": {Raw: []byte(`{"serverMeta": {"turboServer": "https://Turbo_server_URL"}}`)},
					},
				},
			},
		},
	}
}
```

Reply:
Yes, currently, I just use one group of registry and config. We can add another group in the following PR.